src/widgets/widget_definitions.rs

```diff
@@ -146,6 +147,29 @@ pub(super) fn resolve_orientable_attrs(bargs: &mut BuilderArgs, gtk_widget: &gtk
 
 // concrete widgets
 
+/// @widget combo-box
+fn build_gtk_combo_box(bargs: &mut BuilderArgs) -> Result<gtk::ComboBoxText> {
+    let gtk_widget = gtk::ComboBoxText::new();
+    let on_change_handler_id: Rc<RefCell<Option<glib::SignalHandlerId>>> = Rc::new(RefCell::new(None));
+    resolve_block!(bargs, gtk_widget, {
+        // @prop items
+        prop(items: as_vec) {
+            for i in items {
+            gtk_widget.append_text(&i);
+            }
+        },
+        // @prop onchange - runs the code when a item was selected, replacing {} with the item as a string
+        prop(onchange: as_string) {
+            let old_id = on_change_handler_id.replace(Some(
+                gtk_widget.connect_changed(move |gtk_widget| {
+                    run_command(&onchange, gtk_widget.get_active_text().unwrap());
```

Comment:
This `unwrap` will panic if no value is selected. I'd guess that that's a realistic state to be in, thus you should _definitely_ not unwrap here

Reply:
I tested it, and the even only gets triggered, when something is selected.. that means if it has a value, thus the unwrap should *definitely* be ok

Reply:
Still not a fan, why not just do `unwrap_or_default` to have an empty string in case it _does_ happen?

Reply:
understandable

Reply:
fixed with `"".into` in `.unwrap_or()`